modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayTestflowAdministrator.testcase

```diff
@@ -0,0 +1,330 @@
+
+			UserGroup.assignAllMembersCP(userGroupName = "UG UserGroup Name");
+		}
+
+		task ("Go to Testray, Create Project and change pagination") {
+			ApplicationsMenu.gotoSite(site = "Testray 2");
+
+			Testray.createProjectTestray(
+				description = "Projects_1 Description",
+				key_name = "Projects_1 Project");
+		}
+
+		task ("Go to Case Types and add a new case type") {
+			TestrayLayout.clickManageOnSidebar();
+
+			TestrayManage.selectOptionsManage(itemValue = "Case Types");
+
+			TestrayManage.createNewCaseTypes(caseName = "Case_Type");
+
+			TestrayManage.closeModalInOptions();
+		}
+
+		task ("Change pagination and create Team") {
+			Testray.changePagination(
+				numberItems = "20",
+				valuePagination = "150");
+
+			TestrayManage.createTeam(
+				manageOption = "Manage Teams",
+				nameOptions = "Projects_1 Project",
+				version = "Teams_1 Team");
+
+			TestrayManage.closeModalInOptions();
+		}
+
+		task ("Go to component and Add a new Component") {
+			TestrayContextMenu.clickContextMenuProjects(name = "Projects_1 Project");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Manage Components");
+
+			Testray.addNewComponent(
+				componentName = "Components_1 Project",
+				teamName = "Teams_1 Team");
+
+			TestrayManage.closeModalInOptions();
+		}
+
+		task ("Go to project page and add a new routine") {
+			Testray.goToProjectPage(projectName = "Projects_1 Project");
+
+			TestrayRoutines.createRoutine(key_name = "Routines_1 Routine");
+		}
+
+		task ("Go to tab cases and Add a new Case") {
+			Testray.projectTabNavigator(navTabName = "Cases");
+
+			TestrayCases.typeCasesField(
+				caseName = "Cases_1 Case",
+				caseTypeId = "Case_Type",
+				componentId = "Components_1 Project",
+				description = "Cases_1 Case Description",
+				duration = "40",
+				priority = "5",
+				steps = "Auto Test Steps");
+		}
+
+		task ("Go to Routine and Add Environments Factors") {
+			Testray.projectTabNavigator(navTabName = "Routines");
+
+			TestrayContextMenu.clickContextMenuProjects(name = "Routines_1 Routine");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Select Default Environment Factors");
+
+			TestrayBuild.selectDefaultEnvironmentFactors(
+				label = "Application Server",
+				label1 = "Browser",
+				listCategory = "Application Server,Browser",
+				listOption = "JBoss 6.1",
+				listOption1 = "Chrome x");
+		}
+
+		task ("Go to routine and add a new build") {
+			TestrayRoutines.goToRoutine(routineName = "Routines_1 Routine");
+
+			TestrayBuild.createBuild(
+				addVersion = "1.2",
+				caseName = "Cases_1 Case",
+				description = "Builds_1 Project Description",
+				dropdownItem = "New Build",
+				fieldName = "Builds_1 Project",
+				gitHash = "12345",
+				productVersion = "1.2",
+				routine = "Routines_1 Routine");
+		}
+
+		task ("Go to Testray 2 and change the pagination") {
+			ApplicationsMenu.gotoSite(site = "Testray 2");
+
+			TestrayTestflow.goToTestflow();
+		}
+	}
```

Comment:
@joelrocha68 why is this setup so big? Are these tasks really necessary to be executed on each test?

Reply:
Yes, because to create a task is required to have:
1- Project
2- Routine
3- Build
4- Case Results
5- A second user created